transmission_interface/include/transmission_interface/differential_transmission.hpp

```diff
@@ -0,0 +1,403 @@
+ *   (depicted as timing belts in the figure).
+ *  A transmission ratio can take any real value \e except zero. In particular:
+ *     - If its absolute value is greater than one, it's a velocity reducer / effort amplifier, while if its absolute
+ *       value lies in \f$ (0, 1) \f$ it's a velocity amplifier / effort reducer.
+ *     - Negative values represent a direction flip, ie. input and output move in opposite directions.
+ *     - <b>Important:</b> Use transmission ratio signs to match this class' convention of positive actuator/joint
+ *       directions with a given mechanical design, as they will in general not match.
+ *
+ * \note This implementation currently assumes a specific layout for location of the actuators and joint axes which is
+ * common in robotic mechanisms. Please file an enhancement ticket if your use case does not adhere to this layout.
+ *
+ * \ingroup transmission_types
+ */
+class DifferentialTransmission : public Transmission
+{
+public:
+  /**
+   * \param actuator_reduction Reduction ratio of actuators.
+   * \param joint_reduction    Reduction ratio of joints.
+   * \param joint_offset       Joint position offset used in the position mappings.
+   * \pre Nonzero actuator and joint reduction values.
+   */
+  DifferentialTransmission(
+    const std::vector<double> & actuator_reduction,
+    const std::vector<double> & joint_reduction,
+    const std::vector<double> & joint_offset = {0.0, 0.0});
+
+  void configure(
+    const std::vector<JointHandle> & joint_handles,
+    const std::vector<ActuatorHandle> & actuator_handles) override;
+
+  /**
+   * \brief Transform variables from actuator to joint space.
+   * \pre Actuator and joint vectors must have size 2 and point to valid data.
+   *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+   */
+  void actuator_to_joint() override;
+
+  /**
+   * \brief Transform variables from joint to actuator space.
+   * \pre Actuator and joint vectors must have size 2 and point to valid data.
+   *  To call this method it is not required that all other data vectors contain valid data, and can even remain empty.
+   */
+  void joint_to_actuator() override;
+
+  std::size_t num_actuators() const override {return 2;}
+  std::size_t num_joints()    const override {return 2;}
+
+  const std::vector<double> & get_actuator_reduction() const {return actuator_reduction_;}
+  const std::vector<double> & get_joint_reduction()    const {return joint_reduction_;}
+  const std::vector<double> & get_joint_offset()       const {return joint_offset_;}
+
+  /**
+   * \brief Get human-friendly report of handles
+   */
+  std::string get_handles_info() const;
+
+protected:
+  std::vector<double> actuator_reduction_;
+  std::vector<double> joint_reduction_;
+  std::vector<double> joint_offset_;
+
+  std::vector<JointHandle> joint_position_;
```

Comment:
I know this is a larger change. but, would this name be better?

```suggestion
  std::vector<JointInterface> joint_position_;
```

Reply:
I didn't want to mix nomenclature with the interfaces of `ros2_control`. Even though  `JointHandle` and `ActuatorHandle` rely on the same implementation, it is only for convenience purposes, not for compatibility. I'd suggest opening a new issue to discuss this as this PR is not the right place for it.